src/Sylius/Bundle/CoreBundle/Context/ChannelContext.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Context;
+
+use Sylius\Component\Channel\Context\ChannelContext as BaseChannelContext;
+use Sylius\Component\Core\Channel\ChannelContextInterface;
+use Sylius\Component\Core\Channel\ChannelResolverInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+
+/**
+ * Core channel context, which is aware of multiple channels.
+ *
+ * @author Kristian Løvstrøm <kristian@loevstroem.dk>
+ */
+class ChannelContext extends BaseChannelContext implements ChannelContextInterface
+{
+    /**
+     * @var ChannelResolverInterface
+     */
+    protected $channelResolver;
+
+    /**
+     * @var ContainerInterface
+     */
+    protected $container;
+
+    /**
+     * @param ChannelResolverInterface $channelResolver
+     * @param ContainerInterface       $container
+     */
+    public function __construct(ChannelResolverInterface $channelResolver, ContainerInterface $container)
+    {
+        $this->channelResolver = $channelResolver;
+        $this->container = $container;
```

Comment:
Just a random idea - What if we resolve the channel only once during the kernel.request event?